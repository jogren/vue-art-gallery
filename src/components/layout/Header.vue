<template>
  <header class="header">
    <p>About</p>
    <h1>Vue Art Gallery</h1>
  </header>
</template>

<script>
export default {
  name: 'Header'
}
</script>

<style scoped>
  .header {
    height: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
  }
  h1 {
    font-size: 36px;
  }
</style>