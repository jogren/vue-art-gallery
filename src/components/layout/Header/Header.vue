<template>
  <header class="header">
    <h1>Vue Art Gallery</h1>
    <div id="nav">
      <router-link to="/">Home</router-link> |
      <router-link to="/about">About</router-link>
    </div>
  </header>
</template>

<script>
export default {
  name: 'Header'
}
</script>

<style scoped>
  .header {
    height: 120px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
  }
  h1 {
    font-size: 36px;
  }

  #nav {
    padding: 10px;
  }

  #nav a {
    font-weight: bold;
    text-decoration: none;
  }

  #nav a.router-link-exact-active {
    color: #42b983;
  }
</style>