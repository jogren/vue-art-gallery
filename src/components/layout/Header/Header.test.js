import { mount } from '@vue/test-utils'
import Header from './Header'

describe('Header', () => {
  test('is a Vue instance', () => {
    const wrapper = mount(Header)
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    const wrapper = mount(Header)
    expect(wrapper.element).toMatchSnapshot()
  })
})