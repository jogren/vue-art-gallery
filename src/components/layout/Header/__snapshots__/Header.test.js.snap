// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="header"
>
  <h1>
    Vue Art Gallery
  </h1>
   
  <div
    id="nav"
  >
    <router-link
      to="/"
    >
      Home
    </router-link>
     |
    
    <router-link
      to="/about"
    >
      About
    </router-link>
  </div>
</header>
`;
